test/components/form_test.exs

```diff
@@ -123,4 +123,33 @@ defmodule Surface.Components.FormTest do
 
     assert html =~ ~r/Name is invalid/
   end
+
+  test "form generates method input for prop" do
+    html =
+      render_surface do
+        ~H"""
+        <Form for={{ :user }} action="#" method="put" csrf_token="test">
+        </Form>
+        """
+      end
+
+    assert html =~ ~s(<input name="_method" type="hidden" value="put">)
+  end
+
+  test "form generates method input for changeset", %{conn: conn} do
+    assigns = %{
+      "changeset" =>
+        Ecto.Changeset.cast(
+          {%{}, %{name: :string}},
+          %{name: "myname"},
+          [:name]
+        )
+        |> Map.put(:data, %{__meta__: %{state: :loaded}})
```

Comment:
@msaraiva This does not feel right, but couldn't figure out how to create a "loaded" changeset in this test. Any ideas?

Reply:
Hey @miguel-s!

You could define a schema and use `Ecto.put_meta/2`:

```elixir
  defmodule User do
    use Ecto.Schema

    schema "user" do
      field :name, :string
    end
  end

  test "form generates method input for changeset", %{conn: conn} do
    changeset =
      %User{}
      |> Ecto.put_meta(state: :loaded)
      |> Ecto.Changeset.cast(%{name: "myname"}, [:name])

    assigns = %{"changeset" => changeset}

    {:ok, _view, html} = live_isolated(conn, ViewWithForm, session: assigns)

    assert html =~ ~s(><input name="_method" type="hidden" value="put"/>)
  end
```

Can't see any other way.